templates/shared/user/profile_big_avatar.tmpl

```diff
@@ -99,12 +99,12 @@
 			{{if and .IsSigned (ne .SignedUserID .ContextUser.ID)}}
 			<li class="follow">
 				{{if $.IsFollowing}}
-					<form method="post" action="{{.Link}}?action=unfollow&redirect_to={{$.Link}}">
+					<form method="post" action="{{.ContextUser.HomeLink}}?action=unfollow&redirect_to={{$.Link}}">
```

Comment:
Whe you are on stars tab and then clickinng on Follow/UnFollow will redirect to you Repository Tab.

```suggestion
					<form method="post" action="{{.ContextUser.HomeLink}}?action=unfollow&redirect_to={{$.Link}}&tab={{.TabName}}">
```

Reply:
That's an incorrect change and I do not want to make it more complex.

1. Your `tab` is not in the `redirect_to` but it applies to the `href`
1. If you are on other pages, there will be incorrect `tab`
2. The current page could also have other parameters like "page".


There could be a better solution: use "link-action" instead. But at the moment I would avoid changing backend code too much.

Reply:
How do we solve this redirect problem? 
Could you share a short code implementation.